frontend/src/components/ArtifactPreview.tsx

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2021 The Kubeflow Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { useQuery } from 'react-query';
+import { ExternalLink } from 'src/atoms/ExternalLink';
+import { color } from 'src/Css';
+import { Apis } from 'src/lib/Apis';
+import WorkflowParser, { StoragePath } from 'src/lib/WorkflowParser';
+import { stylesheet } from 'typestyle';
+import Banner from './Banner';
+import { ValueComponentProps } from './DetailsTable';
+
+const css = stylesheet({
+  root: {
+    width: '100%',
+  },
+  preview: {
+    maxHeight: 250,
+    overflowY: 'auto',
+    padding: 3,
+    backgroundColor: color.lightGrey,
+  },
+  topDiv: {
+    display: 'flex',
+    justifyContent: 'space-between',
+  },
+  separater: {
+    width: 20, // There's minimum 20px separation between URI and view button.
+    display: 'inline-block',
+  },
+  viewLink: {
+    whiteSpace: 'nowrap',
+  },
+});
+
+export interface ArtifactPreviewProps extends ValueComponentProps<string> {
+  namespace?: string;
+  maxbytes?: number;
+  maxlines?: number;
+}
+
+/**
+ * A component that renders a preview to an artifact with a link to the full content.
+ */
+const ArtifactPreview: React.FC<ArtifactPreviewProps> = ({
+  value,
+  namespace,
+  maxbytes = 255,
+  maxlines = 20,
+}) => {
+  let storage: StoragePath | undefined;
+  if (value) {
+    storage = WorkflowParser.parseStoragePath(value);
+  }
+
+  const { isSuccess, isError, data, error } = useQuery<string, Error>(
+    ['artifact_preview', value],
```

Comment:
nit: you should include all the variables used here to avoid cache collision: "namespace", "maxbytes", "maxlines".

Other variables should be the same most of the time, so they won't affect cache either, but missing "namespace" might mean you can have a collision between different namespaces.

Reply:
That makes sense! I will add all parameters to the query key to avoid collision.